picard/ui/searchdialog.py

```diff
@@ -0,0 +1,466 @@
+        gif_label = QtGui.QLabel(self.progress_widget)
+        movie = QtGui.QMovie(":/images/loader.gif")
+        gif_label.setMovie(movie)
+        movie.start()
+        gif_label.setAlignment(QtCore.Qt.AlignHCenter | QtCore.Qt.AlignTop)
+        layout.addWidget(text_label)
+        layout.addWidget(gif_label)
+        layout.setContentsMargins(1, 1, 1, 1)
+        self.progress_widget.setLayout(layout)
+        self.add_widget_to_center_layout(self.progress_widget)
+
+    def show_error(self, error, show_retry_button=False):
+        """Displays error inside the dialog.
+
+        Args:
+            error -- Error string
+            show_retry_button -- Whether to display retry button or not
+        """
+
+        self.error_widget = QtGui.QWidget(self)
+        self.error_widget.setObjectName("error_widget")
+        layout = QtGui.QVBoxLayout(self.error_widget)
+        error_label = QtGui.QLabel(error)
+        error_label.setWordWrap(True)
+        error_label.setAlignment(QtCore.Qt.AlignCenter)
+        error_label.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse)
+        layout.addWidget(error_label)
+        if show_retry_button:
+            retry_widget = QtGui.QWidget(self.error_widget)
+            retry_layout = QtGui.QHBoxLayout(retry_widget)
+            retry_button = QtGui.QPushButton(_("Retry"), self.error_widget)
+            retry_button.clicked.connect(self.retry)
+            retry_button.setSizePolicy(QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Fixed))
+            retry_layout.addWidget(retry_button)
+            retry_layout.setAlignment(QtCore.Qt.AlignHCenter | QtCore.Qt.AlignTop)
+            retry_widget.setLayout(retry_layout)
+            layout.addWidget(retry_widget)
+        self.error_widget.setLayout(layout)
+        self.add_widget_to_center_layout(self.error_widget)
+
+    def show_table(self, column_headers):
+        """Displays results table inside the dialog."""
+
+        self.table = ResultTable(self.table_headers)
+        self.table.setObjectName("results_table")
+        self.table.cellDoubleClicked.connect(self.row_double_clicked)
+        self.table.horizontalHeader().sectionResized.connect(
+                self.save_table_header_state)
+        self.restore_table_header_state()
+        self.add_widget_to_center_layout(self.table)
+        def enable_loading_button():
+            self.load_button.setEnabled(True)
+        self.table.itemSelectionChanged.connect(
+                enable_loading_button)
+
+    def row_double_clicked(self, row):
+        """Handle function for double click event inside the table."""
+
+        self.load_selection(row)
+        self.accept()
+
+    def network_error(self, reply, error):
+        error_msg = _("<strong>Following error occurred while fetching results:<br></strong>"
+                      "Network request error for %s: %s (QT code %d, HTTP code %s)<br>" % (
```

Comment:
Feel free to add some more newlines to this message. The URLs generated are very long and the errorString can potentially include the whole query sent to the server again if the query could not be parsed. Having all of that on one line with just a bit of wrapping depending on the window size looks a bit ugly.